_extras/guide.md

```diff
@@ -52,7 +53,8 @@ Explain that we use Python because:
 *   It's free.
 *   It has a lot of scientific libraries, and more are constantly being added.
 *   It has a large scientific user community.
-*   It's easier for novices to learn than most of the mature alternatives.
+*   It's easier for novices to learn than most of the mature alternatives, 
+partly due to its syntax that reads more like a human language.
```

Comment:
I agree that we need to explain why [we think] Python is easier to learn [than other programming languages]. However, these bullet points state the reasons why [we think] Python is better suited for teaching basic Programming skills. So, I think it might be better to not explain ourselves right here but move the explanation to the next paragraph. It could start with something like:

> That last point above is important. Code written in Python reads more like a human language. Originally, Software Carpentry was teaching programming using Perl. When we switched to Python, we found..."

What do you guys think?

Reply:
I think it is fine as-is. I think it will be easier to miss the point if it is moved to a separate paragraph. Also, the lead-in to the list starts with "Explain", so I think it is okay to explain in the list.

Reply:
> I think it will be easier to miss the point if it is moved to a separate paragraph.

I'm not suggesting to move the entire point to a separate paragraph -- only its justification / explanation. Keeping only the main points in the list should make them easier to remember. Those who don't read the next paragraph will miss the explanation(s) but not the point(s). 

Speaking of learning... I believe it's the combination of syntax and semantics that makes Python easy to learn: syntax is related to writing proper Python code, and semantics -- to making sense of Python code.

> Also, the lead-in to the list starts with "Explain"

To me this serves as a justification to write bullet points as succinct statements rather than sentences. Explanations that we can provide in the list can only be of limited length and I think we shouldn't limit our explanations by the choice of where we provide them.

Do you think the following would convey the message of the last bullet point?

> Its easy-to-learn syntax and semantics make it ideal for teaching basic programming skills.

Reply:
Taking out from the bullet point the part about Perl as a  justification that Python is easier, now we would be stating Python is "easy-to-learn" which I don't like. What makes for Python being "easy-to-learn syntax and semantics"? It may not be considered easy to someone, especially someone who has no reference to other programming syntax. How are the semantics of Python different than other languages?

Reply:
> Taking out from the bullet point the part about Perl as a justification that Python is easier, now we would be stating Python is "easy-to-learn" which I don't like. 

I guess my last suggestion was not the best and it confused you. Let me try to better explain my concern. 
What we have now in this PR can be represented as:

> we do X because:
> A
> B
> C, partially due to D (E)

This last bullet point seems to be  overloaded with information a little bit, so I propose to say:

> we do X because
> A
> B
> C
>
> C is partially due to D which, in turn, is supported by E.

(A, B, C) together explain why we _teach_ programming using Python.
(C) is the point about Python being easier to _learn_ than other mature alternatives.
I think (C) is important enough to justify spending a paragraph on it.

> How are the semantics of Python different than other languages?

In Python `"1" + 2` is valid syntactically but not semantically.
In Perl, the same expression is valid syntactically and semantically (result is 3).

In Python `"1" + "2"` is valid syntactically and semantically.
In Perl, the identical expression (`"1" + "2"`) is valid syntactically but not semantically (result is 0) and syntax of a semantically valid equivalent is `"1" . "2"`.

So, it's always a combination of syntax and semantics that makes a language intuitive or... not-so-much. I think that the closer the syntax and semantics of a programming language to those of (human language + mathematics), the more intuitive it is. 

> What makes for Python being "easy-to-learn syntax and semantics"? It may not be considered easy to someone, especially someone who has no reference to other programming syntax.

Agreed, this was not the best wording -- we don't have to use it and we can keep "It's easier to learn than most of the mature alternatives". I'm mostly concerned with overloading the last point as I tried to explain in the beginning of this comment.

Reply:
Perhaps you could do a pull request to submit your changes there @maxim-belkin if you are wanting to reword/reorganize beyond what is presented in this pull request . I think we may have lost @ndilekli :/.

Reply:
Hello, sorry that I was out of the loop for a while. I could do another commit, whenever you would like me to. Regarding the Python syntax, I could comment that as someone starting out with Basic, then continuing with Pascal and C, the syntax of Python is indeed much more like reading a text in English. Saying "reads like English" might be an overstatement, but I have seen quite a lot of remarks on that aspect. We might perhaps instead say it is more readable relative to many other programming languages like C and Java. What do you think?

Reply:
Hello, sorry that I was out of the loop for a while. I could do another commit, whenever you would like me to. Regarding the Python syntax, I could comment that as someone starting out with Basic, then continuing with Pascal and C, the syntax of Python is indeed much more like reading a text in English. Saying "reads like English" might be an overstatement, but I have seen quite a lot of remarks on that aspect. We might perhaps instead say it is more readable relative to many other programming languages like C and Java. What do you think?

Reply:
> Hello, sorry that I was out of the loop for a while. 

No worries!

> I could do another commit, whenever you would like me to.

Feel free to update the PR whenever you like if you agree with the comments / suggestions. To sum up:

1. My concern is that we are overloading the last item in the list that explains why we use Python for teaching with information why Python is easier to learn than other (mature) programming languages. Those are related but different topics. So, I propose to move the explanation to the paragraph right after the list. I'm not sure if Lauren (@ldko) agrees with that, so I'd like to clarify this first. If there is no consensus, we might have to leave it as is and discuss this part later in a separate PR or issue.

2. Please have a look at Lauren's suggestion on how to rephrase the first paragraph in introduction: https://github.com/swcarpentry/python-novice-inflammation/pull/837#discussion_r450963457. You can use it "as is" or make modifications you deem important.

>  as someone starting out with Basic, then continuing with Pascal and C

Huh, I took the same path with a quick stop at Visual Basic between Pascal and C (it was interesting to observe how Python adopted and extended Pascal's `:=` assignment operator).


> , the syntax of Python is indeed much more like reading a text in English. Saying "reads like English" might be an overstatement, but I have seen quite a lot of remarks on that aspect. We might perhaps instead say it is more readable relative to many other programming languages like C and Java.

Sure, that'll work too (I think). This way we won't have to go deep into syntax/semantics stuff.

Thank you, Naci!

Reply:
If you all agree about splitting off some of the last item in the list to a paragraph that follows, that is fine with me. Please go ahead.